writers/streamingbatchwriter/unimplemented.go

```diff
@@ -27,3 +27,9 @@ func (UnimplementedDeleteStale) DeleteStale(_ context.Context, ch <-chan *messag
 	}
 	return fmt.Errorf("DeleteStale: %w", plugin.ErrNotImplemented)
 }
+
+type UnimplementedDeleteRecordsBatch struct{}
+
+func (UnimplementedDeleteRecordsBatch) DeleteRecords(_ context.Context, ch <-chan *message.WriteDeleteRecord) error {
+	return fmt.Errorf("DeleteRecordsBatch: %w", plugin.ErrNotImplemented)
```

Comment:
Needs to range over `ch` before returning, else the caller will block.
```suggestion
	// nolint:revive
	for range ch {
	}
	return fmt.Errorf("DeleteRecordsBatch: %w", plugin.ErrNotImplemented)
```